### DISCUSSION TITLE
fmDNS integration with Kubernetes ExternalDNS

### DISCUSSION THREAD

**Discussion by ValeriiVozniuk** [facileManager/facileManager#601] [Q&A] [answered]
Hi,

I was looking for an UI for Bind DNS management, and facileManager/fmDNS looks great for this task, thank you for it :).

After doing the test deployment, I have a task to integrate it with [Kubernetes ExternalDNS](https://kubernetes-sigs.github.io/external-dns) so dynamically created in Kubernetes objects would have their records created automatically in DNS. There is no out-of-box support for fmDNS in it, so I've started to look how to do it manually.

After reading the bugs/some code, I have the following questions regarding API usage in facileManager/fmDNS, and want to confirm if my understanding is correct.

1. To add a DNS record to a zone, I need to run client.php script locally on any client?
2. If client is connected via http/https, can I invoke a web call to add a record, or this is not possible?
3. For DNS zone reload on client, are the same limitations apply? I need to run client.php script locally on each client, or I can do it via web call to each client?
4. Can I trigger all clients reload via web call to the server, or this is not possible too?

I assume, I can get all these answers by thoughtfully reading the source code, but my PHP knowledge is really basic, so it is faster to ask a person who knows the answers :).

**Comment by ValeriiVozniuk** [comment]
One more thing, after reading https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/webhook-provider.md I have a questions: is it hard to implement described there endpoints to allow ExternalDNS to manage records? :)
Per what I see, ExternalDNS can be configured with webhook-provider-url parameter to point to facileManager api endpoint.

**Comment by WillyXJ** [comment, marked as answer]
1. Correct
2. This is currently not possible
3. Zone reloads need to be performed via the WebUI or individually on each client via client.php
4. This is currently not possible through the API
5. This requires a complete rewrite of the API (which is needed anyways) and I'd have to look into ExternalDNS to see what's required. I'd say the level of effort is moderate to hard at this point.

**Comment by WillyXJ** [comment]
Reloading zones from the API is now added in [fmDNS 7.0.0-rc2](https://www.facilemanager.com/modules/fmdns/?t=2) and later.

Reference https://github.com/WillyXJ/facileManager/discussions/589#discussioncomment-11762182 for example usage.
